### PR TITLE
Distribution

### PR DESCRIPTION
**Pull request by ramonsmits**
**Please merge #50 first as this branch is forked from *azure-persistence***

Adds distributor and sender-side distribution.

### CODE REVIEW

File: src/PerformanceTests/Distribution.V5.MsmqDistributor.V5/DistributorProfile.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            var isMaster = string.Equals("master", value, StringComparison.InvariantCultureIgnoreCase);
+
+            if (isDistributor || isMaster)
+            {
+                cfg.RunMSMQDistributor(withWorker: isMaster);
+                return;
+            }
+
+            /*
+            <appSettings>
```
Comment (WilliamBZA) [inline src/PerformanceTests/Distribution.V5.MsmqDistributor.V5/DistributorProfile.cs]: Why is this here?

File: src/PerformanceTests/Distribution.V5.MsmqDistributor.V5/DistributorProfile.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using NServiceBus;
+using Tests.Permutations;
+using Variables;
+
+namespace Distribution.V5.MsmqDistributor.V5
+{
+    using NServiceBus.Config;
```
Comment (WilliamBZA) [inline src/PerformanceTests/Distribution.V5.MsmqDistributor.V5/DistributorProfile.cs]: Some usings inside the NS and some outside?

File: src/PerformanceTests/Distribution.V6.SenderSide/SenderSideProfile.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    public IContext Context { private get; set; }
+
+    public void Configure(EndpointConfiguration cfg)
+    {
+        if (Context.Permutation.ScaleOut != ScaleOut.SenderSide) return;
+        if (Context.Permutation.Transport != Transport.MSMQ) throw new NotSupportedException("SenderSide should only be used with MSMQ");
+
+        var senderSideArgs = this.FetchSetting("SenderSide").Split(';');
+
+        //cfg.ScaleOut().InstanceDiscriminator(senderSideArgs[0]); // Should not be required with MSMQ?
```
Comment (WilliamBZA) [inline src/PerformanceTests/Distribution.V6.SenderSide/SenderSideProfile.cs]: Remove

**Comment by WilliamBZA** [comment]
:+1: once commented out code is removed

File: src/PerformanceTests/Distribution.V5.MsmqDistributor.V5/DistributorProfile.cs (hunk 1 above)
Comment (ramonsmits) [inline src/PerformanceTests/Distribution.V5.MsmqDistributor.V5/DistributorProfile.cs]: @WilliamBZA The config has some settings that you cannot set in code. I left it as reference but if you like I can remove it.
